src/analysis/locate.ml

```diff
@@ -809,6 +809,33 @@ let from_longident ~config ~env ~lazy_trie ~pos nss ml_or_mli ident =
     else
       locate ~config ~ml_or_mli ~path:tagged_path ~lazy_trie ~pos ~str_ident loc
 
+let from_path ~config ~local_defs ~pos ~namespace ml_or_mli path =
+  let str_ident = Path.name path in
+  let browse = Mbrowse.of_typedtree local_defs in
+  let lazy_trie =
+    lazy (Typedtrie.of_browses ~local_buffer:true
+            [Browse_tree.of_browse browse])
+  in
+  let nss_path = Namespaced_path.of_path ~namespace path in
+  let loc = Location.none in
```

Comment:
This is wrong but I'm not sure where to get the right location here.

Reply:
I think you'll want to do something similar to what `Env_lookup.in_namespaces` does: call the correct `Env.find_...` function depending on what the namespace is. (this can probably be an auxiliary function in `Env_lookup`).

Reply:
Actually, I can just get the location directly from the node. Does that work?

This location is only used for:

```
    if not (Utils.is_ghost_loc loc) then Fallback.set loc;
```

Which I don't quite understand. What' a fallback location?

Reply:
The location from the node is the location of the use, not the location of the definition.
Fallback locations are basically … checkpoints in a video game? If you die, you'll start over from here.

Or more seriously: we're trying to walk back from the use to the definition of the identifier, but we might get stuck at any point (because some cmt/cmti might be unavailable). In those cases we don't want to completely fail, we want to go back to the last place we looked at.
For instance if you have
```ocaml
include Foo

let f = something
```
And something comes from `Foo`, but its cmt is missing. Merlin will still jump you to the `include` line, to indicate that `something` comes from there.

Reply:
I see. Thanks for the explanation. I've pushed a commit that attempts to find the right location.